### ISSUE TITLE
Thread-safe access to [OMPromise state]

### ISSUE THREAD

**Issue by iljaiwas** [b52/OMPromises#34]
When I run my app with Xcode's Thread sanitizer enabled, I get an issue reported where read access to the 'state' property is not properly guarded (see attached screenshot). This could be fixed by either declaring the 'state' property as atomic, or by putting the read access within the @synchronized clause. 

Any ideas what would be the best route to fix this?

<img width="1552" alt="Screen Shot 2020-01-11 at 21 22 04" src="https://user-images.githubusercontent.com/2284114/72210280-230efb80-34b9-11ea-9960-7a48d8436ac4.png">

**Comment by aomader** [comment]
You are right, that this is indeed a data race. But the solution is to drop the `self` on that particular line and use the previously requested value.

I do not have the time to fix that right now, but if you could change that and write a PR I would happily merge that.
